docs/reference/scripting/common-script-uses.asciidoc

```diff
@@ -0,0 +1,423 @@
+[[common-script-uses]]
+=== Common scripting use cases
+You can write a script to do almost anything, and sometimes, that's
+the trouble. It's challenging to know what's possible with scripts,
+so the following examples address common uses cases where scripts are
+really helpful.
+
+[[scripting-field-extraction]]
+==== Field extraction
+The goal of this use case is simple; you have fields in your data with a bunch of
+information, but you only want to extract pieces and parts.
+
+There are two options at your disposal:
+
+* <<grok-basics,Grok>> uses a pattern like a regular expression that supports
```

Comment:
It *is* a regular expression, just a kind of unexpected dialect. May be more correct to say "is a regular expression dialect that supports aliased expression reuse." - no need to explain that it sits "on top", I guess, if you say it that way.

Reply:
I'm +1 on that revision, but I still think it's valuable to call out the 1-to-1 mapping of regex in grok. Something like:

>Grok is a regular expression dialect that supports aliased expressions that you can reuse. Because Grok sits on top of regular expressions, any regular expressions are valid in grok as well.